patches/server/1027-Fix-MC-255756-shield-break-block-sounds-not-playing.patch

```diff
@@ -0,0 +1,57 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: roro1506HD <roro1506mcpro@gmail.com>
+Date: Fri, 25 Aug 2023 14:42:54 +0200
+Subject: [PATCH] Fix MC-255756 shield break/block sounds not playing
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+index e11d7283662834047b2ff81a2fd25a4263792deb..7e02de5ae22abeaad6607c5856e8326538716acb 100644
+--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
++++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+@@ -1482,7 +1482,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
+             boolean knockbackCancelled = this.level().paperConfig().environment.disableExplosionKnockback && source.is(DamageTypeTags.IS_EXPLOSION) && this instanceof net.minecraft.world.entity.player.Player; // Paper - Disable explosion knockback
+             if (flag1) {
+                 if (flag) {
+-                    this.level().broadcastEntityEvent(this, (byte) 29);
++                    this.level().playSound(null, this, SoundEvents.SHIELD_BLOCK, SoundSource.PLAYERS, 0.8F, 0.8F + this.level().random.nextFloat() * 0.4F); // Paper - Fixes MC-255756: Play the sound instead of the entity event
+                 } else {
+                     if (!knockbackCancelled) // Paper - Disable explosion knockback
+                     this.level().broadcastDamageEvent(this, source);
+@@ -2225,7 +2225,10 @@ public abstract class LivingEntity extends Entity implements Attackable {
+ 
+             // Apply blocking code // PAIL: steal from above
+             if (event.getDamage(DamageModifier.BLOCKING) < 0) {
+-                this.level().broadcastEntityEvent(this, (byte) 29); // SPIGOT-4635 - shield damage sound
++                // Paper start - Fixes MC-255756: Play the sound instead of the entity event
++                net.minecraft.world.entity.player.Player exemptedPlayer = this instanceof net.minecraft.world.entity.player.Player player ? player : null;
++                this.level().playSound(exemptedPlayer, this, SoundEvents.SHIELD_BLOCK, SoundSource.PLAYERS, 0.8F, 0.8F + this.level().random.nextFloat() * 0.4F);
++                // Paper end
```

Comment:
Why is the player blocking exempted here? broadcastEntityEvent sends it to the current player as well as nearby players.

Reply:
Uhm, I missed the part where it sends it to the player, going to fix that on other places as well